abstractions/typescript/src/authentication/accessTokenProvider.ts

```diff
@@ -0,0 +1,11 @@
+import { RequestInformation } from "../requestInformation";
+
+/**
+ * @interface
+ * An AccessTokenProvider implementation retrieves an access token
+ * to be used by an AuthenticationProvider implementation.
+ * @property {Function} authenticateRequest - The function to authenticate the request.
+ */
+ export interface AccessTokenProvider {
+    getAuthorizationToken: (request?: RequestInformation) => Promise<string>;
```

Comment:
instead of accepting a request, could we change it to accept URL instead so it aligns with other languages and makes the use of this easier for arbitrary requests?

Reply:
(Just a note before we close this) https://github.com/microsoft/kiota/pull/1050 already does this.

Reply:
did you mean for the other languages?

Reply:
Yes.